### PR TITLE
GH-15960 Improve custom metric doc

### PR DESCRIPTION
**Pull request by maurever**
Issue: https://github.com/h2oai/h2o-3/issues/15960

### CODE REVIEW

**Review by wendycwong — CHANGES_REQUESTED** [review]

File: h2o-docs/src/product/data-science/algo-params/custom_metric_func.rst [hunk 1]
```diff
@@ -6,14 +6,61 @@
...
 
-**Note**: This option is only supported in the Python client.
+The Map-Reduce approach is used to calculate the metric. A dataset is broken into multiple chunks and operations are performed on each row of each chunk in the ``map`` function, combined in the ``reduce`` function, and the final value of the metric is returned in the ``metric`` function.
+
+Map
+'''
+
+The ``map`` function defines an action that needs to be performed on each row of the data. Here is a description of each variable for the ``map`` function:
+
+    - ``pred``: a double array storing the model prediction output.  For regresion problems, the prediction value is stored in ``pred[0]``; for classification problems, the prediction value for class ``i`` is stored in ``pred[i]``.
```
[the remark this answers is not in the document]
Comment (maurever) [reply h2o-docs/src/product/data-science/algo-params/custom_metric_func.rst:29]: Thanks. Good catch! :)

File: h2o-docs/src/product/data-science/algo-params/custom_metric_func.rst [hunk 2]
```diff
@@ -6,14 +6,61 @@
...
+
+    - :math:`N` is number of data points
+    - :math:`y(i)` is i-th measurement
+    - :math:`\hat{y}(i)` is prediction of the i-th measurement
+
+You can deconstruct this formula into the map-reduce-value principle where:
+
+- :math:`\Arrowvert y(i) - \hat{y}(i) \Arrowvert ^2` part is done in `map` function
+- :math:`\sum_{i=0}^{N}` part and count N is done in ``reduce`` function with results from ``map`` phase
+- :math:`\sqrt{}` is done in value function from results from ``reduce`` phase
```
[the remark this answers is not in the document]
Comment (maurever) [reply h2o-docs/src/product/data-science/algo-params/custom_metric_func.rst:61]: Not in postGlobal, but in the 'metric' function. Thanks @wendycwong for catch this.

[the remark this answers is not in the document]
Comment (maurever) [reply h2o-docs/src/product/data-science/algo-params/custom_metric_func.rst]: I think I improved the text to be more precious. Am I right now,  @wendycwong?

**Review by wendycwong — DISMISSED** [review]
Veronika:

Thank you for taking care of this one.  Just one minor change request.


Wendy

File: h2o-docs/src/product/data-science/algo-params/custom_metric_func.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (wendycwong) [reply h2o-docs/src/product/data-science/algo-params/custom_metric_func.rst]: Yes and thank you

File: h2o-docs/src/product/data-science/algo-params/custom_metric_func.rst [hunk 3]
```diff
@@ -6,14 +6,61 @@
...
 
-**Note**: This option is only supported in the Python client.
+The Map-Reduce approach is used to calculate the metric. A dataset is broken into multiple chunks and operations are performed on each row of each chunk in the ``map`` function, combined in the ``reduce`` function, and the final value of the metric is returned in the ``metric`` function.
+
+Map
+'''
+
+The ``map`` function defines an action that needs to be performed on each row of the data. Here is a description of each variable for the ``map`` function:
+
+    - ``pred``: a double array storing the model prediction output.  For regresion problems, the prediction value is stored in ``pred[0]``; for classification problems, the prediction value for class ``i`` is stored in ``pred[i+1]``.
```
[the remark this answers is not in the document]
Comment (maurever) [reply h2o-docs/src/product/data-science/algo-params/custom_metric_func.rst]: Thanks @wendycwong!

**Review by wendycwong — APPROVED** [review]

**Pull request merged**